### PR TITLE
Dnd taxonomy resources

### PR DESCRIPTION
**Pull request by kodeebo**
Enable drag and drop of resources in all environments

- Fix index bug when dragging downwards
- Use only icon as drag handle
- Make the title a link to edit the resource
- Added loading indicator when updating the rank, and also when loading topic resources 

Test that:

- [ ] The rank is correctly updated when dragging a resource
- [ ] You cannot drag the resource anywhere else than the icon
- [ ] The links are correct

### CODE REVIEW

File: src/components/MakeDndList.jsx [hunk 1]
```diff
@@ -21,8 +22,9 @@ const MakeDndList = ({ disableDnd, children, onDragEnd }) =>
               'drop-zone',
               snapshot.isDraggingOver ? 'dragging' : '',
             )}>
-            {React.Children.map(children, (child, index) =>
-              child ? (
+            {React.Children.map(children, (child, index) => {
+              if (!child) return null;
```
Comment (chrpeter) [inline src/components/MakeDndList.jsx]: brackets 🥇

File: src/containers/StructurePage/resourceComponents/StructureResources.jsx [hunk 2]
```diff
@@ -143,8 +147,13 @@ export class StructureResources extends React.PureComponent {
...
     } = this.props;
-    const { topicDescription, resourceTypes, topicResources } = this.state;
-
+    const {
+      topicDescription,
+      resourceTypes,
+      topicResources,
+      loading,
+    } = this.state;
+    if (loading) return <Spinner />;
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/StructureResources.jsx]: { } 😏

File: src/containers/StructurePage/resourceComponents/ResourceItems.jsx [hunk 3]
```diff
@@ -57,13 +57,21 @@ class ResourceItems extends React.PureComponent {
...
     if (!destination) return;
     try {
-      const { connectionId, isPrimary } = this.props.resources[source.index];
+      const {
+        connectionId,
+        isPrimary,
+        rank: currentRank,
+      } = this.props.resources[source.index];
       const { rank } = this.props.resources[destination.index];
+      if (currentRank === rank) return;
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/ResourceItems.jsx]: { }

Comment (chrpeter) [reply src/containers/StructurePage/resourceComponents/ResourceItems.jsx]:
```
if (!destination || currentRank === rank) {
  return 
}
```

File: src/containers/StructurePage/resourceComponents/ResourceItems.jsx [hunk 4]
```diff
@@ -57,13 +57,21 @@ class ResourceItems extends React.PureComponent {
   async onDragEnd({ destination, source }) {
     if (!destination) return;
     try {
-      const { connectionId, isPrimary } = this.props.resources[source.index];
+      const {
+        connectionId,
+        isPrimary,
+        rank: currentRank,
+      } = this.props.resources[source.index];
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/ResourceItems.jsx]: destructuring props?

File: src/containers/StructurePage/resourceComponents/ResourceItems.jsx [hunk 5]
```diff
@@ -153,15 +161,25 @@ class ResourceItems extends React.PureComponent {
...
-    const { deleteId, error, filterPickerId, activeFilters } = this.state;
+    const {
+      deleteId,
+      error,
+      filterPickerId,
+      activeFilters,
+      loading,
+    } = this.state;
+
+    if (loading) return <Spinner />;
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/ResourceItems.jsx]: { }

File: src/containers/StructurePage/resourceComponents/ResourceItems.jsx [hunk 6]
```diff
@@ -57,13 +57,21 @@ class ResourceItems extends React.PureComponent {
...
+      if (currentRank === rank) return;
+
+      this.setState({ loading: true });
       await updateTopicResource(connectionId, {
-        rank,
+        rank: currentRank > rank ? rank : rank + 1,
         primary: isPrimary,
       });
-      this.props.refreshResources();
+      await this.props.refreshResources();
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/ResourceItems.jsx]: Se over

Comment (kodeebo) [reply src/containers/StructurePage/resourceComponents/ResourceItems.jsx]: Hva?

**Comment by kodeebo** [comment]
@gunnarvelle sjekk nå

**Pull request merged**